### PR TITLE
Advance payment patch

### PR DESCRIPTION
**Pull request by Olawale1**
Hi

Trust you're doing great. I've added all the basic components for the Advance Payment functionality

Changes to PY file:

1. Added whitelist method for creating Advance Payment entry 
2. Added whitelist method for calculating Total Advance Paid 

Changes to JS file:

1. Added script and made relevant changes for Advance Payment button
2. Added script for calling Total Advance Paid method
3. Added script to auto-fetch Default Advance Account from Company master

Changes to JSON file:

1. Added field for Default Advance Account in Company master
2. Added field for Advance Payment Required in Expense Claim form
3. Added field for Total Advance Paid in Expense Claim form

Changes to Documentation

1. Added 'Managing Advance Payments' section to the Expense Claim documentation

Screenshots of End Results:

company_advance_account

advance_payment_required

make_advance_payment

advance payment patch2

advance payment patch3

Thanks

Kind regards,

### CODE REVIEW

File: erpnext/hr/doctype/expense_claim/expense_claim.js.save [hunk 1]
```diff
@@ -0,0 +1,282 @@
+// Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
```
Comment (rmehta) [inline erpnext/hr/doctype/expense_claim/expense_claim.js.save]: Please remove this file!

**Review by rmehta — CHANGES_REQUESTED** [review]

**Comment by Olawale1** [comment]
I've removed the stray file... not sure how it got there in the first place :)

Thanks

**Comment by mbauskar** [comment]
@Olawale1,

Thank you for contribution,

please check and fix the Travis issues? Closing the PR for now

**Comment by Olawale1** [comment]
Hi @mbauskar 

Thanks for your feedback but I'm not sure the Travis issue had anything to do with my commits. I believe it had something to do with the Selling module

Thanks

**Pull request closed without merging**